Apps/Common/src/Models/Events/DataSourcesBlockedEvent.cs

```diff
@@ -0,0 +1,27 @@
+// -------------------------------------------------------------------------
+//  Copyright © 2019 Province of British Columbia
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+// -------------------------------------------------------------------------
+namespace HealthGateway.Common.Models.Events;
+
+using System.Collections.Generic;
+using HealthGateway.Common.Data.Constants;
+using HealthGateway.Common.Messaging;
+
+/// <summary>
+/// Represents a data source(s) blocked event
+/// </summary>
+/// <param name="Hdid">The hdid associated with the blocked data source(s)</param>
+/// <param name="DataSources">The data sources to block.</param>
+public record DataSourcesBlockedEvent(string Hdid, IEnumerable<DataSource> DataSources) : MessageBase;
```

Comment:
Same as above - I'd recommend considering not to share the DataSource enum in the event signature

Reply:
Changed.